### DISCUSSION TITLE
class형 컴포넌트의 생명주기와 함수형 생명주기는 왜 다를까 ?

### DISCUSSION THREAD

**Discussion by D5ng** [React-Core-Learn/Web-Component#2] [논의]
정답이 아니더라도 괜찮으니, 본인이 생각하는 것들을 말해주세요!

**Comment by Seoin02** [comment]
class형 컴포넌트를 처음 짜봤는데, render, setup 등으로 생명주기를 커스텀 할 수 있는 것 같습니다. 반면, 함수형 컴포넌트는 생명주기를 커스텀 할 수 없기 때문에 다른 거 아닐까요?!

**Reply by D5ng** [reply]
저도 인스턴스 영향이 있다고 생각해요. 클래스는 관련한 메서드를 가지고있기 때문에 생명주기를 세세하게 컨트롤할 수 있다고 생각합니다. 

반면 함수형은 함수가 종료되면 소멸되기때문에 클래스처럼 세세한 컨트롤은 힘들다고 보는데, 문득 궁금한 것이 함수안의 헬퍼함수들을 만든다면 가능하지 않을까? 라는 생각이 들어요. 

함수의 재사용성을 위해 일부러 저렇게 안만든것 같기도하고, 헬퍼 함수들을 직접 만들어야하다보니 함수형 프로그래밍과 맞지않은것 같기도하네요(선언적이지 않음)
